server/src/main/java/com/cloud/network/vpc/NetworkACLServiceImpl.java

```diff
@@ -103,12 +104,15 @@ public class NetworkACLServiceImpl extends ManagerBase implements NetworkACLServ
 
     @Override
     public NetworkACL createNetworkACL(final String name, final String description, final long vpcId, final Boolean forDisplay) {
-        final Account caller = CallContext.current().getCallingAccount();
-        final Vpc vpc = _entityMgr.findById(Vpc.class, vpcId);
-        if (vpc == null) {
-            throw new InvalidParameterValueException("Unable to find VPC");
+        if (vpcId != 0) {
+            final Account caller = CallContext.current().getCallingAccount();
+            final Vpc vpc = _vpcDao.findById(vpcId);
```

Comment:
why does this need to change from `_entityMgr.findById(Vpc.class, vpcId)` to `_vpcDao.findById(vpcId)`? I a sure someone implemented this on purpose at some point.

Reply:
The method `_entityMgr.findById` will call the method `_vpcDao.findById`. Is there a reason to use `entityManager` in this case?

Reply:
nah